### ISSUE TITLE
`extend_with_edges`: allow supplying node weights as alternative to `Default`-construction

### ISSUE THREAD

**Issue by sanmai-NL** [bluss/petgraph#189]
E.g, [`Graph::extend_with_edges` in 0.4.10](https://docs.rs/petgraph/0.4.10/petgraph/graph/struct.Graph.html#method.extend_with_edges) states that node weights are `Default`-constructed. 
If there is fundamental reason why so, should the API be expanded to allow for supplying the node weights in the edge list as well?

**Comment by makoConstruct** [comment]
I think we're having the same problem? I'm getting a compilation error `the trait bound `NodeData: std::default::Default` is not satisfied` (where NodeData is what I specified for N), when I'm just trying to add edges between indexes of pre-existing nodes. If I understand the graph implementation correctly, adding edges between indexes should not create any new nodes, so N should not need to be default constructible and I should not be getting this error.

**Comment by makoConstruct** [comment]
It looks like it requires default construction so that, if given node indexes for nodes that don't exist, it can create them. Would the project accept a pull request for a version of the method that just panics if it's given an index to a node that doesn't exist?

**Comment by sanmai-NL** [comment]
No, your issue is a bit different. Maybe the name `extend_with_edges` is a bit confusing. It really means: just extend the graph, here’s a the minimal info needed, an edge list to use for that. The addition of an edge is just one of its effects. What you want is: add an edge, here are the source and target vertices. Use [`add_edge`](https://docs.rs/petgraph/0.4.10/petgraph/stable_graph/struct.StableGraph.html#method.add_edge).

**Comment by bluss** [comment]
help wanted: design an API that allows this. Whether it should change the existing extend_with_edges method or not is to decide in the design.

**Comment by RaoulLuque** [comment]
This feature is still missing and seems like a great addition :) So, if anyone wants to have a go, feel free 🦕
